app/actions/client.py

```diff
@@ -0,0 +1,84 @@
+import httpx
+import backoff
+import logging
+
+from app.actions.configurations import (
+    AuthenticateConfig,
+    FetchSamplesConfig,
+    PullObservationsConfig
+)
+from app.services.errors import ConfigurationNotFound
+from app.services.utils import find_config_for_action
+from app.services.state import IntegrationStateManager
+
+
+state_manager = IntegrationStateManager()
+logger = logging.getLogger(__name__)
+
+
+def get_auth_config(integration):
+    # Look for the login credentials, needed for any action
+    auth_config = find_config_for_action(
+        configurations=integration.configurations,
+        action_id="auth"
+    )
+    if not auth_config:
+        raise ConfigurationNotFound(
+            f"Authentication settings for integration {str(integration.id)} "
+            f"are missing. Please fix the integration setup in the portal."
+        )
+    return AuthenticateConfig.parse_obj(auth_config.data)
+
+
+def get_fetch_samples_config(integration):
+    # Look for the login credentials, needed for any action
+    fetch_samples_config = find_config_for_action(
+        configurations=integration.configurations,
+        action_id="fetch_samples"
+    )
+    if not fetch_samples_config:
+        raise ConfigurationNotFound(
+            f"fetch_samples settings for integration {str(integration.id)} "
+            f"are missing. Please fix the integration setup in the portal."
+        )
+    return FetchSamplesConfig.parse_obj(fetch_samples_config.data)
+
+
+def get_pull_config(integration):
+    # Look for the login credentials, needed for any action
+    pull_config = find_config_for_action(
+        configurations=integration.configurations,
+        action_id="pull_observations"
+    )
+    if not pull_config:
+        raise ConfigurationNotFound(
+            f"pull_config settings for integration {str(integration.id)} "
+            f"are missing. Please fix the integration setup in the portal."
+        )
+    return PullObservationsConfig.parse_obj(pull_config.data)
+
+
+@backoff.on_predicate(backoff.constant, jitter=None, interval=60)
+async def get_positions_list(integration, config):
```

Comment:
Looks like the `config` parameter isn't used in `get_positions_list()`, it could be removed. Also imo `get_positions_list()` doesn't need to receive an integration object, it could receive just the username and password if that's everything needed to make that api call and get positions. Also, helper methods for parsing configs would be better placed in `handlers.py` or maybe `configurations.py`, so that `client.py` only contains code related to interacting with the third-party api (think it as if you wanted to turn the client.py into an independant package in the future)